### PR TITLE
RTP Header Processing

### PR DESCRIPTION
**Pull request by fluffy**

### CODE REVIEW

File: draft-uberti-avtcore-cryptex.md [hunk 1]
```diff
@@ -45,6 +45,22 @@ Signaling
...
+When sending an RTP packet that requires any header extensions to a
+destination that has negotiated header encryption, the header extensions
+be encapsulated inside a {{RFC8285}} header extension.
+
+The 16 bit RTP header extension tag MUST be changed from 0xBEDE to
+0xC0DE to indicate that it will be encrypted.
+
+## Receiving
+
+After decrypting and authenticating the packet, if there is an RPT
```
Comment (juberti) [inline draft-uberti-avtcore-cryptex.md:59]: Here I would just say that 0xC0DE extensions are processed exactly like the one-byte headers defined in S 4.2 of 8285. We also need to define a magic tag for two-byte headers  (see S 4.3 of 8285, perhaps, 0xC02, or 0x200)

File: draft-uberti-avtcore-cryptex.md [hunk 2]
```diff
@@ -45,6 +45,22 @@ Signaling
 RTP Header Processing
 =====================
 
+## Sending
+
+When sending an RTP packet that requires any header extensions to a
```
Comment (juberti) [inline draft-uberti-avtcore-cryptex.md:50]: As noted below, we should point people at 8285 and simply say what is different than the normal processing.

File: draft-uberti-avtcore-cryptex.md [hunk 3]
```diff
@@ -54,6 +70,8 @@ Security Considerations
 IANA Considerations
 ===================
 
+TODO: allocate RTP header extension code 0xC0DE to this spec
```
Comment (murillo128) [inline draft-uberti-avtcore-cryptex.md:73]: Is this really needed? I have failed to find where the 0xBEDE or 0x100 values are registered in IANA for 1 bytes or 2 bytes extension.

File: draft-uberti-avtcore-cryptex.md (hunk 1 above)
Comment (murillo128) [reply draft-uberti-avtcore-cryptex.md:59]: what do we do with the appbits of the 2 bytes header extension? I think we should just not allow to use them with cryptex.

File: draft-uberti-avtcore-cryptex.md [hunk 4]
```diff
@@ -45,6 +45,22 @@ Signaling
...
+be encapsulated inside a {{RFC8285}} header extension.
+
+The 16 bit RTP header extension tag MUST be changed from 0xBEDE to
+0xC0DE to indicate that it will be encrypted.
+
+## Receiving
+
+After decrypting and authenticating the packet, if there is an RPT
+header extension with a 16 bit RTP header extension tag of 0xC0DE, it
+MUST be changed from 0xC0DE to 0xBEDE so that it can be processed as a
```
Comment (murillo128) [inline draft-uberti-avtcore-cryptex.md:61]: I think this in an implementation detail, we should only specify the format that goes over the wire. If the implementation takes the normal profile id as input and changes it when doing the encryption to the encrypted id (and viceversa on decryption) or not is something we should not cover in the spec.

Comment (ibc) [reply draft-uberti-avtcore-cryptex.md:61]: Honestly I think this should be done by the srtp encrypt/decrypt call. I see zero reasons to change the content of the extensions block without also replacing the proper id value that indicates how such a content is.

Comment (ibc) [reply draft-uberti-avtcore-cryptex.md:61]: This is like if you mangle a packet by decrypting its payload without changing the encrypted bit/flag/id. No sense IMHO. Just update it so the new packet represents full valid information.

Comment (JonathanLennox) [reply draft-uberti-avtcore-cryptex.md:61]: My suggestion would be to add an informational note, that implementations may wish to perform the reverse transformation on decryption.  I agree we should only be specifying the wire format, but this is an obvious implementation technique.

Comment (ibc) [reply draft-uberti-avtcore-cryptex.md:61]: Ok, sorry, I understood that it was the header extension id reverse transformation should be done by the application (e.g. the RTP endpoint). Let me fix my concern: it should be done at SRTP stack level when invoking the encrypt/decrypt calls on a packet.

Comment (fluffy) [reply draft-uberti-avtcore-cryptex.md:61]: Agree with Inaki should be done in libSRTP before the app gets the packet. I'll try and make that clearer. What i was really trying to say is after you inverse the encryption transform, this is processed just like normal header extension.

File: draft-uberti-avtcore-cryptex.md (hunk 3 above)
Comment (fluffy) [reply draft-uberti-avtcore-cryptex.md:170]: I will remove this

**Comment by fluffy** [comment]
due to the way the rebase showed up, I'm just going to close and start a new PR for this but I think I addressed the comments so will be good to have a look  at new one

**Pull request closed without merging**